File: server/app/models/complaint.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from datetime import datetime
from ..database import Base


class Complaint(Base):
    __tablename__ = "complaints"

    id = Column(Integer, primary_key=True, index=True)
    student_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    canteen_id = Column(Integer, ForeignKey("canteens.id"), nullable=True)
    meal_id = Column(Integer, ForeignKey("meals.id"), nullable=True)
    order_id = Column(Integer, ForeignKey("orders.id"), nullable=True)

    message = Column(Text, nullable=False)
    status = Column(String(30), default="pending")  # pending, reviewing, resolved
    created_at = Column(DateTime, default=datetime.utcnow)

    student = relationship("User", back_populates="complaints")
    canteen = relationship("Canteen", back_populates="complaints")
    meal = relationship("Meal", back_populates="complaints")
    order = relationship("Order", back_populates="complaints")

File: app/lib/models/complaint.dart
// lib/models/complaint.dart
class Complaint {
  final int id;
  final int studentId;
  final int? canteenId;
  final int? mealId;
  final int? orderId;
  final String message;
  final String status;

  Complaint({
    required this.id,
    required this.studentId,
    this.canteenId,
    this.mealId,
    this.orderId,
    required this.message,
    required this.status,
  });

  factory Complaint.fromJson(Map<String, dynamic> json) {
    return Complaint(
      id: json['id'] as int,
      studentId: json['student_id'] as int,
      canteenId: json['canteen_id'] as int?,
      mealId: json['meal_id'] as int?,
      orderId: json['order_id'] as int?,
      message: json['message'] as String,
      status: json['status'] as String,
    );
  }
}
